runtime/include/chpl-mem.h

```diff
@@ -153,7 +153,8 @@ void* chpl_mem_array_alloc(size_t nmemb, size_t eltSize, c_sublocid_t subloc,
 
     p = NULL;
     *callAgain = false;
-    if (size >= chpl_comm_regMemAllocThreshold()) {
+    if (size >= chpl_comm_regMemAllocThreshold()
+        && size >= 2 * chpl_comm_regMemHeapPageSize()) {
```

Comment:
I'd prefer to put the `size >= chpl_comm_regMemAllocThreshold()  && size >= 2 * chpl_comm_regMemHeapPageSize()` code in some helper function instead of duplicating the logic in alloc and free

Reply:
Done (5a1b0ec7).